network/libp2p_impl.go

```diff
@@ -95,12 +106,16 @@ func (impl *libp2pDataTransferNetwork) openStream(ctx context.Context, id peer.I
 			return s, err
 		}
 
-		nAttempts := b.Attempt()
-		if nAttempts == impl.maxStreamOpenAttempts {
-			return nil, xerrors.Errorf("exhausted %d attempts but failed to open stream, err: %w", impl.maxStreamOpenAttempts, err)
+		// b.Attempt() starts from zero
+		nAttempts := b.Attempt() + 1
+		if nAttempts >= impl.maxStreamOpenAttempts {
```

Comment:
Not really related to this PR, but I am not sure why these counters are `float64`s (both `maxStreamOpenAttempts` and `b.Attempt()`?

Reply:
Good question, I was wondering the same thing - basically it's because of the library we're using for the restart measurement